include/deal.II/lac/read_write_vector.h

```diff
@@ -610,7 +610,7 @@ namespace LinearAlgebra
     /**
      * Pointer to the array of local elements of this vector.
      */
-    std::unique_ptr<Number[]> values;
+    std::unique_ptr<Number[], decltype (free) *> values = std::unique_ptr<Number[], decltype (free) *> (nullptr, free);
```

Comment:
isn't the type already complete determined by the rvalue/xvalue? I.e. can we use
```
auto values = std::unique_ptr(nullptr, free);
```
instead?

Reply:
Ah, this is a member declaration. Sorry for the confusion.
At least a while ago we decided to not use the copy assignment operator for member initialization. Is this still the case?
What about you initialize with `values(nullptr, free)` in the constructor?